hummingbot/market/market_base.pyx

```diff
@@ -138,15 +157,15 @@ cdef class MarketBase(NetworkIterator):
         """
         exchange_limits = self.get_exchange_limit_config(self.name)
 
-        for asset_name, total_balance in self._account_balances.items():
-            if asset_name.upper() in exchange_limits:
-                asset_limit = Decimal(exchange_limits[asset_name.upper()])
-                self._account_balances.update({asset_name: min(total_balance, asset_limit)})
-
         for asset_name, available_balance in self._account_available_balances.items():
             if asset_name.upper() in exchange_limits:
                 asset_limit = Decimal(exchange_limits[asset_name.upper()])
-                self._account_available_balances.update({asset_name, min(available_balance, asset_limit)})
+
+                active_asset_balance = self.in_flight_asset_balances(self.in_flight_orders).get(asset_name.upper(), s_decimal_0)
+                current_asset_available_balance = self._account_available_balances.get(asset_name.upper(), s_decimal_0)
+
+                new_asset_available_balance = min(Decimal(current_asset_available_balance + active_asset_balance), asset_limit)
```

Comment:
this should be 
new_asset_available_balance = min(current_asset_available_balance, max(asset_limit - active_asset_balance, s_decimal_0))